src/us/kbase/common/service/JsonClientCaller.java

```diff
@@ -357,108 +357,125 @@ public static AuthToken requestTokenFromKBase(
 			conn.setFixedLengthStreamingMode(size);
 		}
 		// Write real data into http output stream
-		writeRequestDataStatic(method, arg, conn.getOutputStream(), id, context);
+		try (final OutputStream out = conn.getOutputStream()) {
+			writeRequestDataStatic(method, arg, out, id, context);
+		}
+		catch (IOException ignore){
```

Comment:
I'm not a fan of ignoring exceptions - is there a reason not to pass it up the stack? Presumably this shouldn't happen unless the stream errors on a close, which indicates a problem.